### PR TITLE
Gather usages of JDK dependent OS beans into OsBeanUtil

### PR DESCRIPTION
**Pull request by lutovich**
Currently many different places use proprietary implementations of `java.lang.management.OperatingSystemMXBean` to get information about machine's memory and file descriptors. Some can only work for Oracle JDK and some are able to handle both Oracle JDK and IBM JDK.

This commit introduces `OsBeanUtil` that is able to give out information about memory and file descriptors and work on both JDKs. It also makes all relevant places use this util.

### CODE REVIEW

**Comment by chrisvest** [comment]
Nice.

File: community/kernel/src/main/java/org/neo4j/helpers/OsBeanUtil.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.helpers;
```
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/helpers/OsBeanUtil.java]: Should this be in `org.neo4j.helpers`?  That package is a public thing and something we may have to take extra care for when changing. Would probably be better to have in `org.neo4j.kernel.impl.util` or similar.

Comment (lutovich) [inline community/kernel/src/main/java/org/neo4j/helpers/OsBeanUtil.java]: Good point. Will move it to `org.neo4j.kernel.impl.util`. Would be good to introduce some dedicated module for things like this in 3.0.

**Comment by davidegrohmann** [comment]
@lutovich needs rebase

**Comment by lutovich** [comment]
@davidegrohmann done

**Pull request merged**